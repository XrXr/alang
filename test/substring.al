main :: proc () {
	match("so", "jjjjjjjjsojjjjj")
	match("so", "jjjjjjjjs")
	match("so", "jjojjjsjjjs")
}

// I know the code can be simpler, but this is a regression test
// so please don't edit this proc
match :: proc (sub string, whole string) {
	match := false
	for i := 0..whole.length-sub.length {
		substringMatch := true
		for j := 0..sub.length-1 {
			if @(sub.data + j) != @(whole.data + i + j) {
				substringMatch = false
			}
		}
		if substringMatch {
			match = true
			break
		}
	}
	puts(sub)
	puts(" is ")
	if !match {
		puts("not ")
	}
	puts("a substring of ")
	puts(whole)
	puts("\n")
}
