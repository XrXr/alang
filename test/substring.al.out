so is a substring of jjjjjjjjsojjjjj
so is not a substring of jjjjjjjjs
so is not a substring of jjojjjsjjjs
